### PR TITLE
[NativeAOT-LLVM] Don't use `mmap` in the frozen object manager and enable writeable data

### LINKED ISSUE #2476: [NativeAOT-LLVM] Remove unix file Internal\Runtime\FrozenObjectHeapManager.Unix.cs
We should replace the use of this file as it uses mmap which is not correctly implemented for Wasm.

### LINKED ISSUE #2477: [NativeAOT-LLVM] Turn on TargetSupportsWritingFieldData for Wasm
There is an error when turning on `TargetSupportsWritingFieldData` for Wasm, we should investigate what it is because it would be nice to take advantage of the PreInit code in Ilc.

See https://github.com/dotnet/runtimelab/pull/2473/files/a910fc31dbf068ab26b9645cb153dfbf98cf4320..7f0d1e96556ca65c82f324d984a461dcd920a6f0#r1440477152

And reenable the test `TestTypeHandles.Run();`  (https://github.com/dotnet/runtimelab/pull/2473/files/a910fc31dbf068ab26b9645cb153dfbf98cf4320..7f0d1e96556ca65c82f324d984a461dcd920a6f0#r1440492528)

### PR DESCRIPTION
**Pull request by SingleAccretion**
Fixes #2476.
Fixes #2477.

The latter part part of this change will be upstreamed.

### CODE REVIEW

**Comment by SingleAccretion** [comment]
@dotnet/nativeaot-llvm

**Review by yowl — APPROVED** [review]
Thanks!

File: src/coreclr/tools/aot/ILCompiler.LLVM/CodeGen/LLVMObjectWriter.cs [hunk 1]
```diff
@@ -313,6 +315,16 @@ private void EmitObjectNode(ObjectNode node, ObjectData nodeContents)
             }
         }
 
+        private static bool IsSupportedRelocType(ObjectNode node, RelocType type)
+        {
+            if (node is StackTraceMethodMappingNode)
+            {
+                // Stack trace metadata uses relative pointers, but is currently unused.
```
Comment (jkotas) [inline src/coreclr/tools/aot/ILCompiler.LLVM/CodeGen/LLVMObjectWriter.cs:322]: Do we have an issue about deleting it or making it work?

Comment (SingleAccretion) [reply src/coreclr/tools/aot/ILCompiler.LLVM/CodeGen/LLVMObjectWriter.cs:322]: Yep - this is #2404.

**Pull request merged**